jaxrs/WFLY-10646_EAP7-1034_multivalue_param.adoc

```diff
@@ -0,0 +1,381 @@
+      response.close();
+   }
+....   
+
+is
+
+....
+array segments: 2
+list segments: 2
+....
+
+An alternative is to use a wildcard template parameter. For example, the output of calling testWildcardArray() and testWildcardList() in
+
+....
+@Path("")
+public static class TestResource {
+
+   @GET
+   @Path("{segments:.*}/array")
+   public Response getWildcardArray(@PathParam("segments") PathSegment[] segments) {
+      System.out.println("array segments: " + segments.length);
+      return Response.ok().build();
+   }
+   
+   @GET
+   @Path("{segments:.*}/list")
+   public Response getWildcardList(@PathParam("segments") List<PathSegment> segments) {
+      System.out.println("list segments: " + segments.size());
+      return Response.ok().build();
+   }
+   
+...
+
+   @Test
+   public void testWildcardArray() throws Exception {
+      Invocation.Builder request = client.target("http://localhost:8081/a/b/c/array").request();
+      Response response = request.get();
+      response.close();
+   }
+   
+   @Test
+   public void testWildcardList() throws Exception {
+      Invocation.Builder request = client.target("http://localhost:8081/a/b/c/list").request();
+      Response response = request.get();
+      response.close();
+   }
+....
+
+is
+....
+array segments: 3
+list segments: 3
+....
+
+=== Extension to ParamConverter semantics
+
+In the JAX-RS semantics, a ParamConverter is supposed to convert a single String that represents an individual object. RESTEasy extends the semantics to allow a ParamConverter to parse the String representation of multiple objects and generate a List<T>, Set<T>, SortedSet<T>, array, or, indeed, any multivalued data structure whatever. First, consider the resource
+
+....
+@Path("queryParam")
+public static class TestResource {
+
+   @GET
+   @Path("")
+   public Response conversion(@QueryParam("q") List<String> list) {
+      return Response.ok(stringify(list)).build();
+   }
+}
+
+private static <T> String stringify(List<T> list) {
+   StringBuffer sb = new StringBuffer();
+   for (T s : list) {
+      sb.append(s).append(',');
+   }
+   return sb.toString();
+}
+....
+
+Calling TestResource as follows, using the standard notation,
+
+....
+@Test
+public void testQueryParamStandard() throws Exception {
+   ResteasyClient client = new ResteasyClientBuilder().build();
+   Invocation.Builder request = client.target("http://localhost:8081/queryParam?q=20161217&q=20161218&q=20161219").request();
+   Response response = request.get();
+   System.out.println("response: " + response.readEntity(String.class));
+}
+....
+
+results in
+
+`response: 20161217,20161218,20161219,`
+   
+
+Suppose, instead, that we want to use a comma separated notation. We can add
+
+....
+public static class MultiValuedParamConverterProvider implements ParamConverterProvider
+
+   @SuppressWarnings("unchecked")
+   @Override
+   public <T> ParamConverter<T> getConverter(Class<T> rawType, genericType, Annotation[] annotations) {
```

Comment:
genericType doesn't have class, it leads to compilation error, same in http://docs.jboss.org/resteasy/docs/3.6.0.Final/userguide/html_single/

Reply:
https://github.com/resteasy/Resteasy/pull/1639 ok?

Reply:
@marekkopecky, ah, yes, that should be "Type genericType".

Reply:
@asoldano +1